uint8 armor_id
uint8 HP_deduction_reason